modules/thunder_gqls/src/Wrappers/EntityListResponse.php

```diff
@@ -2,29 +2,65 @@
 
 namespace Drupal\thunder_gqls\Wrappers;
 
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
 use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\graphql\GraphQL\Buffers\EntityBuffer;
 use GraphQL\Deferred;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * The thunder entity list response class.
  */
-class EntityListResponse implements EntityListResponseInterface {
+class EntityListResponse implements EntityListResponseInterface, ContainerInjectionInterface {
 
   /**
    * The query interface.
    *
    * @var \Drupal\Core\Entity\Query\QueryInterface
    */
-  protected $query;
+  private QueryInterface $query;
+
+  /**
+   * The entity buffer.
+   *
+   * @var \Drupal\graphql\GraphQL\Buffers\EntityBuffer
+   */
+  private EntityBuffer $buffer;
 
   /**
    * EntityListResponse constructor.
    *
+   * @param \Drupal\Core\Entity\Query\QueryInterface|\Drupal\graphql\GraphQL\Buffers\EntityBuffer $parameter
+   *   The query or buffer parameter.
+   */
+  public function __construct(QueryInterface|EntityBuffer $parameter) {
+    if ($parameter instanceof QueryInterface) {
+      // phpcs:ignore
+      @trigger_error('Calling the constructor with the query parameter is deprecated in Thunder 7.3.3 it will be remove in Thunder 8.0. Use service injection instead of directly instantiating and then use ::setQuery() instead.', E_USER_DEPRECATED);
+      $this->setQuery($parameter);
+      return;
+    }
+    $this->buffer = $parameter;
+  }
```

Comment:
Normally this would look something like this:

```suggestion
  public function __construct(QueryInterface|EntityBuffer $buffer) {
    if ($buffer instanceof QueryInterface) {
      // phpcs:ignore
      @trigger_error('Calling the constructor with the query parameter is deprecated in Thunder 7.3.3 it will be remove in Thunder 8.0. Use service injection instead of directly instantiating and then use ::setQuery() instead.', E_USER_DEPRECATED);
      $this->setQuery($buffer);
      $buffer = \Drupal::service('graphql.buffer.entity');
    }
    $this->buffer = $buffer;
  }
```